config/application.rb

```diff
@@ -33,6 +33,19 @@ class Application < Rails::Application
 
     config.autoload_paths << Rails.root.join('lib')
 
+    # custom exceptions handling to render responses based on controller
+    # uncaught API errors will now render as JSON responses w/ 500 status
+    # normal controller errors will show 500 page, except in development environment (will show normal error page)
+    # lambda allows for inspecting environment, which will include request parameters
+    config.exceptions_app = lambda do |env|
```

Comment:
this is some gnarly stuff--nice work digging it all out!
It seems odd to have a switch based on environment in this file.  

If possible, I'd lean towards just redefining `config.exceptions_app` in `config/environments/development.rb` so that we're not mixing environments in config files

Reply:
7f87f88